src/main/java/com/google/step/finscholar/servlets/CollegeServlet.java

```diff
@@ -29,8 +33,18 @@
 /** Servlet that returns colleges and their associated data. */
 @WebServlet("/college-data")
 public class CollegeServlet extends HttpServlet {
+  private Firestore database;
+  private Gson gson;
 
-  private final Gson gson = new Gson();
+  @Override
+  public void init(ServletConfig config) throws ServletException {
+    try {
+      database = FirestoreClient.getFirestore(FirebaseAppManager.getApp());
```

Comment:
I think this method is called only when the servelet comes online, meaning that the credentials will only be initialized once. Have you considered moving it to doGet instead?

Reply:
Yeah that works.